ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java

```diff
@@ -205,8 +211,22 @@ private void registerSchema(
         final ParsedSchema parsedSchema = translator.toParsedSchema(
             PersistenceSchema.from(schema, serdeFeatures)
         );
-
-        srClient.register(subject, parsedSchema);
+        if (parsedSchema instanceof ProtobufSchema) {
+          final ProtobufSchema resolved = AbstractKafkaProtobufSerializer.resolveDependencies(
+              srClient,
+              true,
+              true,
```

Comment:
nit: the arg for `useLatestVersion` is not used because `autoRegisterSchema` is true, but it would probably be best to pass `false` anyway (since `false` is the default for `useLatestVersion` in general).  So the args should read `srClient, true, false, true`.

Reply:
@agavra , sure I've filed https://confluentinc.atlassian.net/browse/DGS-1285 to track.  The method recursively registers referenced schemas.